src/js/plugins/tts/AbstractTTSEngine.js

```diff
@@ -91,28 +79,20 @@ export default class AbstractTTSEngine {
         this.playing = true;
         this.opts.onLoadingStart();
 
-        /** @type {AsyncStream<PageChunk>} */
-        this.chunkStream = AsyncStream.range(leafIndex, numLeafs-1)
-        .map(this.fetchPageChunks.bind(this))
-        .buffer(2)
-        .flatten()
-        .map(chunk => {
-            this.opts.onLoadingStart();
-            chunk.sound = this.createSound(chunk);
-            chunk.sound.rate = this.playbackRate;
-            chunk.sound.voice = this.voice;
-            chunk.sound.load(() => this.opts.onLoadingComplete());
-            return chunk;
+        this._chunkIterator = new PageChunkIterator(numLeafs, leafIndex, {
+            server: this.opts.server,
+            bookPath: this.opts.bookPath,
+            pageBufferSize: 5,
         });
 
         this.step();
         this.events.trigger('start');
     }
 
     stop() {
-        this.activeSound.stop();
+        if (this.activeSound) this.activeSound.stop();
```

Comment:
They should be redundant, but I'd like them to happen anyways; there are a lot of async things happening, so it's very possible for the system to fall out of sync. Setting these variables ensures that the state variables reflect the true state of the system.

Reply:
hmmm, to me. `true` state would only fire the event if it actually stopped.

>  there are a lot of async things happening, so it's very possible for the system to fall out of sync.

This statement makes me think that it should only fire if we stopped it.  But, also, I am here as a soundboard for you. :) carry on, if you think the system needs this echo at all times

Reply:
I actually agree with you 100% :P this is an example of slightly "paranoid" code (which is usually a bad thing). In this case, because the underlying WebSpeech API is _really_ unreliable/finicky is why this code is a little ham-fisted :P